packages/taco/src/conditions/base/address-allowlist.ts

```diff
@@ -0,0 +1,27 @@
+import { Condition } from '../condition';
+import {
+  AddressAllowlistConditionProps,
+  addressAllowlistConditionSchema,
+  AddressAllowlistConditionType,
+} from '../schemas/address-allowlist';
+import { OmitConditionType } from '../shared';
+export { AddressAllowlistConditionProps, AddressAllowlistConditionType };
+
+/**
+ * A condition that checks if a wallet address is in a list of allowed addresses.
+ */
+export class AddressAllowlistCondition extends Condition {
+  constructor(value: OmitConditionType<AddressAllowlistConditionProps>) {
+    super(addressAllowlistConditionSchema, {
+      conditionType: AddressAllowlistConditionType,
+      ...value,
+    });
+  }
+
+  /**
+   * This condition requires wallet authentication to verify addresses
+   */
+  override requiresAuthentication(): boolean {
+    return true;
+  }
```

Comment:
Not sure if this is a question or a suggestion, but what do you think about not overriding requiresAuthentication()? Since we are going to require the decryptor's authentication for this condition, we can just add it to the constructor.

This looks to me more appropriate. Or at least, closer to the logic of the Condition class.

Of course, I don't think your approach is wrong and it's totally ok to me if you keep it.

```suggestion
  constructor(value: OmitConditionType<AddressAllowlistConditionProps>) {
    super(addressAllowlistConditionSchema, {
      conditionType: AddressAllowlistConditionType,
      parameters: [USER_ADDRESS_PARAM_DEFAULT]
      ...value,
    });
  }
```

Reply:
I would lean toward adding `parameters: [USER_ADDRESS_PARAM_DEFAULT]` if that would be needed inside the condition somewhere else. But I think adding it just for the sake of making requiresAuthentication may cause confusion on what is this param used for exactly.

Reply:
You will need to explicitly add `:userAddress` somewhere in the condition schema for the node to properly process the condition. We have made the conscious decision that any requirement for use of :userAddress should be explicit and not implicit, since it complicates the logic both on the client and in the node i.e. you shouldn't even need to override `requiresAuthentication` as long as `:userAddress` is in the condition.

For eg. ERC20Balance - https://github.com/nucypher/taco-web/blob/main/packages/taco/src/conditions/predefined/erc20.ts#L14.

That being said, it doesn't have to be:
```typescript
parameters: [USER_ADDRESS_PARAM_DEFAULT]
```
That was only because the rpc call takes an array of parameters. 

Instead you can just define a field as part of the schema, that only allows for the user address context variable to be provided.

```typescript
conditionType: z.literal(AddressAllowlistConditionType),
requesterAddress: UserAddressSchema,
allowedAddresses: z.array(...)
```